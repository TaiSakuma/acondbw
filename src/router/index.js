import Vue from "vue";
import VueRouter from "vue-router";
import Home from "@/views/Home.vue";
import ProductTop from "@/views/ProductTop.vue";

import ALL_MAPS from "@/graphql/AllMaps.gql";

Vue.use(VueRouter);

const routes = [
  {
    path: "/",
    name: "home",
    component: Home,
  },
  {
    path: "/about",
    name: "about",
    component: () =>
      import(/* webpackChunkName: "about" */ "@/views/About.vue"),
  },
  {
    path: "/simulations",
    component: ProductTop,
    props: {
      title: "Simulations",
      icon: "mdi-creation",
      itemPageName: "SimulationItem",
    },
    children: [
      {
        path: "",
        name: "SimulationList",
        component: () =>
          import(
            /* webpackChunkName: "simulations" */ "@/views/SimulationList.vue"
          ),
      },
      {
        path: "item/:name",
        name: "SimulationItem",
        component: () =>
          import(
            /* webpackChunkName: "simulations" */ "@/views/SimulationItem.vue"
          ),
      },
    ],
  },
  {
    path: "/maps",
    component: ProductTop,
    props: { title: "Maps", icon: "map", itemPageName: "MapItem" },
    children: [
      {
        path: "",
        name: "MapList",
        component: () =>
          import(/* webpackChunkName: "maps" */ "@/views/MapList.vue"),
        props: {
          productTypeName: { singular: "map", plural: "maps" },
          query: ALL_MAPS,
          queryName: "allMaps",
          productIdFieldName: "mapId",
          productItemCard: "MapItemCard",
          productAddForm: "MapAddForm",
          disableAdd: process.env.VUE_APP_ACONDBW_MAP_MUTATION_DIALOG != "true",
        },
      },
      {
        path: "item/:name",
        name: "MapItem",
        component: () =>
          import(/* webpackChunkName: "maps" */ "@/views/MapItem.vue"),
      },
    ],
  },
  {
    path: "/beams",
    component: ProductTop,
    props: {
      title: "Beams",
      icon: "mdi-spotlight-beam",
      itemPageName: "BeamItem",
    },
    children: [
      {
        path: "",
        name: "BeamList",
        component: () =>
          import(/* webpackChunkName: "beams" */ "@/views/BeamList.vue"),
      },
      {
        path: "item/:name",
        name: "BeamItem",
        component: () =>
          import(/* webpackChunkName: "beams" */ "@/views/BeamItem.vue"),
      },
    ],
  },
];

const router = new VueRouter({
  mode: "history",
  base: process.env.BASE_URL,
  routes,
});

export default router;
