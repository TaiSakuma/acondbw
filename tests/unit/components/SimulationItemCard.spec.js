import Vue from "vue";
import VueRouter from "vue-router";
import Vuetify from "vuetify";
import { mount, createLocalVue } from "@vue/test-utils";

import SimulationItemCard from "@/components/SimulationItemCard.vue";
import router from "@/router";

Vue.use(Vuetify);
Vue.use(VueRouter);

describe("SimulationItemCard.vue", () => {
  let localVue;
  let vuetify;

  function createWrapper({ loading = false, propsData } = {}) {
    const mutate = jest.fn();
    let wrapper = mount(SimulationItemCard, {
      localVue,
      router,
      vuetify,
      mocks: {
        $apollo: {
          queries: {
            node: {
              loading: loading
            }
          },
          mutate
        }
      },
      propsData: {
        simulationName: "lat20200201",
        ...propsData
      }
    });

    return wrapper;
  }

  const node = {
    id: "U2ltdWxhdGlvbjoxMDAx",
    productId: "1001",
    name: "xyz-s1234-20200101",
    datePosted: "2019-03-15",
    mapper: "abc-def",
    note: "- note 1\n- **marked**"
  };

  beforeEach(function() {
    localVue = createLocalVue();
    vuetify = new Vuetify();
  });

  it("loading", async () => {
    const loading = true;
    const wrapper = createWrapper({ loading });
    await Vue.nextTick();
    expect(wrapper.find(".v-progress-circular").exists()).toBe(true);
  });

  it("error", async () => {
    const wrapper = createWrapper();
    wrapper.setData({
      error: "Error: cannot load data"
    });
    await Vue.nextTick();
    expect(wrapper.text()).toContain("Error: cannot load data");
  });

  it("none", async () => {
    const wrapper = createWrapper();
    await Vue.nextTick();
    expect(wrapper.text()).toContain("Nothing to show here");
  });

  it("match snapshot", async () => {
    const wrapper = createWrapper();
    wrapper.setData({
      node: node
    });
    await Vue.nextTick();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it.each([
    [true, true, false],
    [true, false, true],
    [false, true, true],
    [false, false, true]
  ])(
    "collapsed - {collapsible: %p, collapsed: %p, visible: %p}",
    async (collapsible, collapsed, visible) => {
      const wrapper = createWrapper({
        propsData: { collapsed: collapsed, collapsible: collapsible }
      });
      wrapper.setData({
        node: node
      });
      await Vue.nextTick();
      expect(wrapper.find(".collapsible").isVisible()).toBe(visible);
    }
  );
});
