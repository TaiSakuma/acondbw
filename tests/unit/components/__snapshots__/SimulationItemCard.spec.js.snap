// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimulationItemCard.vue match snapshot 1`] = `
<div class="simulation-item-card" style="position: relative;" simulationname="lat20200201">
  <div class="v-card v-card--hover v-card--outlined v-sheet theme--light" style="max-width: 980px;">
    <div class="" style="cursor: default;">
      <div class="container pa-0 container--fluid">
        <div class="row ma-0 px-0">
          <div class="py-0 col-md-4 col-10 order-1">
            <div class="caption grey--text">Name</div>
            <div class="font-weight-medium primary--text"><span><a href="/simulations/item/xyz-s1234-20200101" class="">xyz-s1234-20200101</a></span></div>
          </div>
          <div class="py-0 col-md-4 col-6 order-3">
            <div class="caption grey--text">Date posted</div>
            <div>2019-03-15</div>
          </div>
          <div class="py-0 col-md-2 col-6 order-4">
            <div class="caption grey--text">Mapper</div>
            <div>abc-def</div>
          </div>
          <div class="py-0 order-md-5 col col-2 order-2 align-self-end">
            <div class="row px-1 py-0 align-start justify-end">
              <!----> <span><button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-dots-vertical theme--light"></i></span></button>
              <div class="v-menu">
                <!---->
              </div>
              </span></div>
          </div>
        </div>
        <div class="row mx-0 mb-3 px-0 collapsible">
          <div class="py-0 col-md-8 offset-md-4 col-12">
            <div class="caption grey--text">Note</div>
            <div>
              <ul>
                <li>note 1</li>
                <li>note 2</li>
              </ul>
            </div>
          </div>
          <div class="py-0 col col-12">
            <div class="row">
              <div class="py-0 col-md-4 order-md-0 col-12 order-2 align-self-end"><span class="grey--text" style="font-size: 65%;">Data ID: </span></div>
              <div class="py-0 col-md-8 col-12 order-1">
                <div class="caption grey--text">Beams</div>
                <!---->
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div><button type="button" class="v-btn v-btn--absolute v-btn--flat v-btn--icon v-btn--round theme--light v-size--default" style="top: -15px; right: -10px;"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-nut theme--light grey--text text--lighten-1" style="font-size: 12px;"></i></span></button>
    <div class="v-menu">
      <!---->
    </div>
  </div>
</div>
`;
